modin/engines/base/io/io.py

```diff
@@ -519,21 +520,26 @@ def read_hdf(
         **kwargs,
     ):  # noqa: PR01
         ErrorMessage.default_to_pandas("`read_hdf`")
-        return cls.from_pandas(
-            pandas.read_hdf(
-                path_or_buf,
-                key=key,
-                mode=mode,
-                columns=columns,
-                errors=errors,
-                where=where,
-                start=start,
-                stop=stop,
-                iterator=iterator,
-                chunksize=chunksize,
-                **kwargs,
-            )
+        modin_store = isinstance(path_or_buf, HDFStore)
+        if modin_store:
+            path_or_buf._return_modin_dataframe = False
+        df = pandas.read_hdf(
+            path_or_buf,
```

Comment:
In some cases, `modin.HDFStore` object passed in `pandas.read_hdf` function that does not count on it.

As a solution for the case, it is proposed to disable conversion to modin objects.

Reply:
I would prefer to avoid return any pandas objects. Modin and pandas objects do not play well together. Worst case we should throw an error, but ideally this is fixable to return a Modin dataframe.

Reply:
Pandas object will be return only before `cls.from_pandas(df)` call, in which `df` will be cast to Modin dataframe.